### PR TITLE
05 09 19 meal index

### PR DESCRIPTION
**Pull request by Mackenzie-Frey**
### Meal Index API endpoint
This pull request implements functionality for the following user story:

### User Story
As a user sending a `GET` request to `/api/v1/meals`, it returns all the meals in the database along with their associated foods

If successful, this request will return a response in the following format:

```
[
    {
        "id": 1,
        "name": "Breakfast",
        "foods": [
            {
                "id": 1,
                "name": "Banana",
                "calories": 150
            },
            {
                "id": 6,
                "name": "Yogurt",
                "calories": 550
            },
            {
                "id": 12,
                "name": "Apple",
                "calories": 220
            }
        ]
    },
    {
        "id": 2,
        "name": "Snack",
        "foods": [
            {
                "id": 1,
                "name": "Banana",
                "calories": 150
            },
            {
                "id": 9,
                "name": "Gum",
                "calories": 50
            },
            {
                "id": 10,
                "name": "Cheese",
                "calories": 400
            }
        ]
    },
    {
        "id": 3,
        "name": "Lunch",
        "foods": [
            {
                "id": 2,
                "name": "Bagel Bites - Four Cheese",
                "calories": 650
            },
            {
                "id": 3,
                "name": "Chicken Burrito",
                "calories": 800
            },
            {
                "id": 12,
                "name": "Apple",
                "calories": 220
            }
        ]
    },
    {
        "id": 4,
        "name": "Dinner",
        "foods": [
            {
                "id": 1,
                "name": "Banana",
                "calories": 150
            },
            {
                "id": 2,
                "name": "Bagel Bites - Four Cheese",
                "calories": 650
            },
            {
                "id": 3,
                "name": "Chicken Burrito",
                "calories": 800
            }
        ]
    }
]
```

- [x] Successful response includes `200` status.
- [x] If no meals are found in the database it returns a  `404` status with the message `{"error": "There are no meals in the database"}`
- [x] Unsuccessful response returns a `500` internal server error. 

Endpoint is fully tested for:
- [x] 200 status
- [x] Returning an array with the correct number of objects and that each object has the attributes `id`, `name`, and `calories`.

-------------------
- [x] Endpoint has been documented in the README
- [ ] Endpoint has been refactored

Functional endpoint in Postman:
image

Tests for `meal_index.spec.js` are passing:
image

### CODE REVIEW

File: routes/api/v1/meals.js [hunk 1]
```diff
@@ -7,22 +7,61 @@ const fetch = require('node-fetch');
...
+        attributes: ['id', 'name', 'calories'],
+        through: {
+          attributes: []
+        }
+      }]
+    })
+    .then(meals => {
+      if (!meals) {
+        res.status(404).send({
+          error: 'There are no meals in the database.'
```
Comment (nagerz) [inline routes/api/v1/meals.js:53]: Have you tested this? I believe no meals would result in an empty array and wouldn't hit here (you're not using a 'where' in the findAll as opposed to we did for the findOne). Don't think you need the if statement.

File: specs/endpoints/meal_index.spec.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  });
+  beforeEach(() => {
+    shell.exec('npx sequelize db:migrate:undo:all')
+    shell.exec('npx sequelize db:seed:undo:all')
+    shell.exec('npx sequelize db:migrate')
+    shell.exec('npx sequelize db:seed:all')
+  });
+  afterEach(() => {});
+
+  describe('Test GET /api/v1/meals', () => {
```
Comment (nagerz) [inline specs/endpoints/meal_index.spec.js:17]: This describe should be moved above the setup/teardown and that should be replaced with the update version. I will make this update along with my other test updates.

Comment (Mackenzie-Frey) [reply specs/endpoints/meal_index.spec.js:17]: Sounds good.

File: routes/api/v1/meals.js (hunk 1 above)
Comment (Mackenzie-Frey) [reply routes/api/v1/meals.js:53]: I'll hop on testing this.

**Pull request merged**